pyro/distributions/util.py

```diff
@@ -1,24 +1,29 @@
 from __future__ import absolute_import, division, print_function
 
-import functools
-import os
-
 import torch
 import torch.nn.functional as F
 from torch.autograd import Variable
 
-# TODO Decide based on torch.__version__ once torch.distributions matures.
-USE_TORCH_DISTRIBUTIONS = int(os.environ.get('PYRO_USE_TORCH_DISTRIBUTIONS', 0))
-
 
-def torch_wrapper(pyro_dist):
+def copy_docs_from(source_class):
     """
-    Decorator for optional wrappers around torch.distributions classes.
+    Decorator to copy class and method docs from source to destin class.
     """
-    if USE_TORCH_DISTRIBUTIONS:
-        return lambda wrapper: functools.wraps(pyro_dist)(wrapper)
-    else:
-        return lambda wrapper: pyro_dist
+
+    def decorator(destin_class):
+        destin_class.__doc__ = source_class.__doc__
+        for name in dir(destin_class):
+            if name.startswith('_'):
+                continue
+            try:
+                source_attr = getattr(source_class, name)
+                destin_attr = getattr(destin_class, name)
+                destin_attr.__doc__ = source_attr.__doc__
```

Comment:
Should we check if `destin_attr.__doc__` already exists and only change if it doesn't, so that the subclasses don't have the docstrings replaced by the base class?

Reply:
Good idea, I'll do that.

Reply:
Done.